### PR TITLE
Rainbow crayon + color-selectable crayons

### PR DESCRIPTION
**Pull request by vulppine**
## About the PR 

Adds the rainbow crayon - which, in turn, adds the ability to have color selectable crayons. The rainbow crayon comes with 30 charges, instead of the regular 5 found in regular crayons.

**Screenshots**

image

**Changelog**

:cl:
- add: Adds the rainbow crayon. Honk!

### CODE REVIEW

File: Content.Server/Crayon/CrayonSystem.cs [hunk 1]
```diff
@@ -108,6 +123,29 @@ private void OnCrayonBoundUI(EntityUid uid, CrayonComponent component, CrayonSel
...
+            {
+                var hexColor = Color.TryFromHex(component._color);
+                if (hexColor != null)
+                {
+                    component._color = args.Color;
+                }
+            }
+        }
+
         Dirty(component);
```
Comment (PaulRitter) [inline Content.Server/Crayon/CrayonSystem.cs:149]: only dirty if you actually change the color

**Review by PaulRitter — CHANGES_REQUESTED** [review]
empty file on client `Content.Client/Crayon/CrayonDecalVisualizer.cs`

god the colorfield on crayoncomp should just be typed color, suppose i'll go fix that soon

**Comment by vulppine** [comment]
> empty file on client `Content.Client/Crayon/CrayonDecalVisualizer.cs`
> 
> god the colorfield on crayoncomp should just be typed color, suppose i'll go fix that soon

already deleted in this PR, and yeah, i was wondering why the color field on crayons was a string

**Comment by PaulRitter** [comment]
> already deleted in this PR

its still there

> yeah, i was wondering why the color field on crayons was a string

previous crimes against humanity

**Comment by vulppine** [comment]
> > already deleted in this PR
> 
> its still there
> 
> > yeah, i was wondering why the color field on crayons was a string
> 
> previous crimes against humanity

image

buh

**Pull request merged**